ArmPlatformPkg/MemoryInitPei/MemoryInitPeim.c

```diff
@@ -114,8 +115,16 @@ InitializeMemory (
   // Declare the UEFI memory to PEI
   //
 
-  // In case the firmware has been shadowed in the System Memory
-  if ((FdBase >= SystemMemoryBase) && (FdTop <= SystemMemoryTop)) {
+  // MU_CHANGE START: Allow platform to customize initial memory region.
+  UefiMemorySize = FixedPcdGet32 (PcdSystemMemoryUefiRegionSize);
+  if (ArmPlatformGetPeiMemory (&UefiMemoryBase, &UefiMemorySize)) {
+    // Check the Firmware does not intersect with the provided memory region.
+    ASSERT ((FdBase < UefiMemoryBase) || (FdBase >= (UefiMemoryBase + UefiMemorySize)));
+    ASSERT ((FdTop <= UefiMemoryBase) || (FdTop > (UefiMemoryBase + UefiMemorySize)));

```

Comment:
in this case should we do anything besides ASSERT? i.e. in the release case is falling through the desired behavior or should we revert to the old behavior?